src/mtev_conf.c

```diff
@@ -94,6 +96,114 @@ static xmlDocPtr master_config = NULL;
 static int config_include_cnt = -1;
 static int backingstore_include_cnt = -1;
 
+struct param_entry {
+  char *name;
+  char *xpath;
+  mtev_param_type_t ptype;
+  void *memory;
+  mtev_param_validator_t validate;
+};
+
+
+mtev_boolean
+mtev_conf_register_global_param(const char *name, const char *xpath,
+                                mtev_param_type_t ptype, void *mem,
+                                mtev_param_validator_t validate) {
+  struct param_entry *p = calloc(1, sizeof(*p));
+  p->name = strdup(name);
+  if(xpath) p->xpath = strdup(xpath);
+  p->ptype = ptype;
+  p->memory = mem;
+  p->validate = validate;
+  if(mtev_hash_store(&global_param_sets, p->name, strlen(p->name), p))
+    return mtev_true;
+  free(p->name);
+  free(p->xpath);
+  free(p);
```

Comment:
I worry that it might be too much change as code various places has interesting defaulting logic.  I was going for plumbing not fixtures: a way to update params online.